x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractFunctionTestCase.java

```diff
@@ -642,6 +642,46 @@ public interface ExpectedEvaluatorToString {
         Matcher<String> evaluatorToString(int nullPosition, TestCaseSupplier.TypedData nullData, Matcher<String> original);
     }
 
+    /**
+     * Modifies suppliers to generate BytesRefs with random offsets.
+     */
+    protected static List<TestCaseSupplier> randomizeBytesRefsOffset(List<TestCaseSupplier> testCaseSuppliers) {
```

Comment:
Could you try moving this into the ctor of `TypedData`? That'd get it everywhere for "free" which seems worth it.

Reply:
Copying from the original PR:

About the TypedData constructor, I would avoid it, as I could imagine some test manually adding offsets themselves for some usecase (?), as well as making things too complex (Now a plain constructor would also modify your data).
Not sure, but I feel like there's more control this way

Added:
I was thinking that would be interesting to replace the long chaing of supplier modifiers (failing types, nulls, etc) into a simpler "recommendedChain()". With that, we would easily apply any check everywhere. And for tests that don't want something, they can always do it the old way